src/Components/Aspire.NATS.Net/AspireNatsClientExtensions.cs

```diff
@@ -92,7 +92,27 @@ NatsConnection Factory(IServiceProvider provider)
                 throw new InvalidOperationException($"NATS connection string not found: {connectionName}");
             }
 
-            options = options with { Url = settings.ConnectionString };
+            var redactedConnectionString = settings.ConnectionString;
+
+            var uriBuilder = new UriBuilder(settings.ConnectionString);
+
+            if (uriBuilder.UserName is { Length: > 0 } username && uriBuilder.Password is { Length: > 0 })
+            {
+                options = options with
+                {
+                    AuthOpts = options.AuthOpts with
+                    {
+                        Username = uriBuilder.UserName,
+                        Password = uriBuilder.Password
+                    }
+                };
+
+                // redact password so it is not visible in logs
+                uriBuilder.Password = "***";
```

Comment:
I think this should be logged as an issue on the NATS.Net library and fixed there. This doesn't make sense at this level.

Reply:
Since the NATS.Net library currently just ignores the user/password info in the url we could also just remove both here completely before passing it to the library. I just thought maybe seeing that it is a authenticated connection could be interesting in the logs, so I redacted instead of removed. For the health check connection we could also remove that user information before passing it to the connection.

Reply:
ok I think I am understanding this now. so there is no other way to pass the user pass but in the connection string?

Reply:
Yes, connection string or extra environmental variables.